test/UniswapV4DeployerCompetition.t.sol

```diff
@@ -69,7 +74,52 @@ contract UniswapV4DeployerCompetitionTest is Test {
         competition.updateBestAddress(salt);
     }
 
-    function testUpdateBestAddressOpen(bytes32 salt) public {
+    function test_updateBestAddress_reverts_InvalidSigner(bytes32 salt) public {
+        vm.assume(bytes20(salt) != bytes20(0));
+        vm.assume(bytes20(salt) != bytes20(winner));
+
+        vm.expectRevert(abi.encodeWithSelector(IUniswapV4DeployerCompetition.InvalidSender.selector, salt, winner));
+        vm.prank(winner);
+        competition.updateBestAddress(salt);
+    }
+
+    function test_updateBestAddress_equalSalt_reverts_WorseAddress(bytes32 salt) public {
+        vm.assume(salt != bytes32(0));
+        console2.logBytes32(salt);
+        salt = (salt & mask20bytes) | bytes32(bytes20(winner));
```

Comment:
I think maybe prefer having the address in the top bits rather than bottom? Just makes it a prefix and incrementing the lower bits feels more natural

Reply:
maybe i'm wrong but i think because bytes are left-aligned this is already in the top bits. Let me check 👀

Reply:
confirmed this btw - address is left aligned